### PR TITLE
chore: merge 3.6 into main

### PR DESCRIPTION
**Pull request by tlm**
Forward merge of changed from 3.6 into main.
    
     Conflicts:
            api/controller/controller/controller.go
            api/controller/controller/legacy_test.go
            apiserver/facades/client/controller/controller.go
            cmd/juju/commands/mocks/controller_mock.go
            cmd/juju/commands/package_test.go
            cmd/juju/commands/upgradecontroller.go
            cmd/juju/commands/upgrademodel_test.go
            cmd/juju/controller/destroy.go
            cmd/juju/controller/destroy_test.go
            cmd/juju/controller/export_test.go
            cmd/juju/controller/kill_test.go
            cmd/juju/controller/showcontroller.go
            cmd/juju/controller/showcontroller_test.go
            go.mod
            go.sum
            internal/worker/uniter/hook/hook.go
            internal/worker/uniter/hook/hook_test.go
            internal/worker/uniter/runner/context/context.go
            internal/worker/uniter/runner/context/env_test.go
            worker/uniter/runner/context/contextfactory_test.go

### CODE REVIEW

**Comment by hpidcock** [comment]
` Error:     | internal/worker/uniter/runner/context/context.go:310:2: field `baseUpgradeTarget` is unused (unused)`

File: internal/charm/hooks/hooks.go [hunk 1]
```diff
@@ -108,7 +109,6 @@ func StorageHooks() []Kind {
 }
 
 var workloadHooks = []Kind{
-	PebbleChangeUpdated,
 	PebbleCustomNotice,
 	PebbleReady,
```
Comment (hpidcock) [inline internal/charm/hooks/hooks.go:113]: Need `PebbleCheckFailed` and `PebbleCheckRecovered`.
See https://github.com/juju/charm/commit/c09ff56304f49b8d98af09ea3a568f8547bdfc5d

File: internal/charm/hooks/hooks.go [hunk 2]
```diff
@@ -141,7 +141,7 @@ func (kind Kind) IsStorage() bool {
 // IsWorkload returns whether the Kind represents a workload hook.
 func (kind Kind) IsWorkload() bool {
 	switch kind {
-	case PebbleChangeUpdated, PebbleCustomNotice, PebbleReady:
+	case PebbleCustomNotice, PebbleReady:
```
Comment (hpidcock) [inline internal/charm/hooks/hooks.go:144]: Need `PebbleCheckFailed` and `PebbleCheckRecovered`.
See https://github.com/juju/charm/commit/c09ff56304f49b8d98af09ea3a568f8547bdfc5d

**Review by hpidcock — APPROVED** [review]
A few changes.

File: internal/charm/hooks/hooks.go (hunk 2 above)
Comment (tlm) [reply internal/charm/hooks/hooks.go]: Done & Done

File: internal/charm/hooks/hooks.go (hunk 1 above)
Comment (tlm) [reply internal/charm/hooks/hooks.go:115]: Done & Done

**Pull request merged**